### PR TITLE
Feature/letra 95 - Add remaining tests for component buttons, add router tests, and add store tests

### PR DESCRIPTION
**Pull request by tbarland77**
Hi @jayehernandez, it took a while but I have another PR for #124. I made the following changes:

- Added tests for the remaining component buttons (random button and translate button)

- Removed tests that were only testing the component name as those were causing deprecation warnings

- Added tests for the router

- Added tests for the store

- Refactored/modified the store slightly to export the actions and mutations so I could easily import them in the test file

There's a decent bit here especially in the store test file as there was quite a bit of mocking I had to do. That file could probably be a bit cleaned up as well. Anyway, I just wanted to go ahead and open this PR so you could have a look. Let me know what you think!

### CODE REVIEW

File: client/test/src/store/index.test.js [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+      it('commits set daily data', () => {
+        actions.saveDailyData({ commit }, data);
+        expect(commit).toHaveBeenCalledWith('setDailyData', data);
+      });
+    });
+    describe('getLanguageOptions', () => {
+      let languageOptions;
+      beforeEach(() => {
+        languageOptions = [
+          'japanese',
```
Comment (jayehernandez) [inline client/test/src/store/index.test.js:257]: Is there a reason why this is written 11 times? :)

**Comment by nicolas-alv3** [comment]
Hi!, Do you need more test? (In adittion to the ones that @tbarland77 did)

File: client/test/src/store/index.test.js (hunk 1 above)
Comment (tbarland77) [reply client/test/src/store/index.test.js]: At the time I did that so I could test the if statement here: 
` if (!!data && data.length === currentLanguagesCount)`
I probably could have made that more obvious by not using the same language 11 times though. :sweat_smile: Anyway I adjusted it to set the length of the array to be currentLanguagesCount in the test. This way too as more languages get added the test should not need to be updated.

**Comment by tbarland77** [comment]
Thanks! Glad to hear you're learning more about tests through my PR's. It definitely took a while testing the store but I learned quite a bit especially in regards to testing promises. Hopefully, before long now Letra will have full code coverage. :smiley:

File: client/test/src/store/index.test.js (hunk 1 above)
Comment (jayehernandez) [reply client/test/src/store/index.test.js]: Got this!

**Comment by jayehernandez** [comment]
Yes to full code coverage. 🥳 I think this PR is good to go, thank you once again @tbarland77!

**Pull request merged**